pandora-client-web/src/components/header/leaveButton.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { ReactElement, createContext, useCallback, useContext, useState } from 'react';
+import logoutIcon from '../../assets/icons/logout.svg';
+import { HeaderButton } from './HeaderButton';
+import { useShardConnectionInfo } from '../gameContext/shardConnectorContextProvider';
+import { usePlayer, usePlayerData, usePlayerState } from '../gameContext/playerContextProvider';
+import { useCurrentAccount, useDirectoryConnector } from '../gameContext/directoryConnectorContextProvider';
+import { EMPTY, GetLogger, IChatRoomFullInfo } from 'pandora-common';
+import { Button } from '../common/button/button';
+import { useLogout } from '../../networking/account_manager';
+import { useCharacterRestrictionsManager, useChatRoomInfo } from '../gameContext/chatRoomContextProvider';
+import { PlayerCharacter } from '../../character/player';
+import { toast } from 'react-toastify';
+import { TOAST_OPTIONS_ERROR } from '../../persistentToast';
+import { ModalDialog } from '../dialog/dialog';
+import { Column } from '../common/container/container';
+import './leaveButton.scss';
+import { HoverElement } from '../hoverElement/hoverElement';
+
+const leaveButtonContext = createContext(() => { /** noop */ });
+
+export function LeaveButton() {
+	const [show, setShow] = useState(false);
+	const closeDialog = useCallback(() => setShow(false), []);
+
+	return (
+		<leaveButtonContext.Provider value={ closeDialog }>
+			<HeaderButton icon={ logoutIcon } iconAlt='Leave' onClick={ () => setShow(true) } title='Leave' />
+			{
+				show ? <DialogLeave /> : null
+			}
+		</leaveButtonContext.Provider>
+	);
+}
+
+function DialogLeave(): ReactElement {
+	const closeDialog = useContext(leaveButtonContext);
+
+	return (
+		<ModalDialog>
+			<Column className='LeaveDialog'>
+				<ChatRoomLeave />
+				<CharacterLeave />
+				<AccountLeave />
+				<Button onClick={ closeDialog }>Cancel</Button>
+			</Column>
+		</ModalDialog>
+	);
+}
+
+function ChatRoomLeave(): ReactElement | null {
+	const player = usePlayer();
+	const room = useChatRoomInfo();
+
+	if (!player || !room) {
+		return null;
+	}
```

Comment:
I think instead of not showing it, it is much better to show the area, but instead of button show that you are not in a room (which will later on turn into that you are in a room that you cannot leave - your personal room)